cli/src/lib/token-storage.ts

```diff
@@ -0,0 +1,211 @@
+import envPaths from "env-paths";
+import {
+  chmodSync,
+  existsSync,
+  mkdirSync,
+  readFileSync,
+  unlinkSync,
+  writeFileSync,
+} from "fs";
+import { join } from "path";
+
+/**
+ * Token data structure stored on disk
+ */
+export interface StoredToken {
+  /** The session token for API authentication */
+  sessionToken: string;
+  /** When the token expires (ISO string) */
+  expiresAt: string;
+  /** User information */
+  user: {
+    id: string;
+    email: string | null;
+    name: string | null;
+  };
+  /** When the token was stored (ISO string) */
+  storedAt: string;
+}
+
+/**
+ * Name of directory to store state/config/cache
+ */
+const DIR_PREFIX = "tambo";
+
+/**
+ * Get OS-specific paths for tambo state storage
+ * Uses XDG Base Directory Specification on Linux
+ *
+ * Data paths (for auth tokens and persistent data):
+ * - macOS: ~/Library/Application Support/tambo/ (or $XDG_DATA_HOME/tambo/)
+ * - Linux: ~/.local/share/tambo/ (or $XDG_DATA_HOME/tambo/)
+ * - Windows: %LOCALAPPDATA%/tambo/
+ */
+function getDir(type: "cache" | "config" | "data"): string {
+  const paths = envPaths(DIR_PREFIX, { suffix: "" });
+  const envKey = `XDG_${type.toUpperCase()}_HOME`;
+  const xdgDir = process.env[envKey];
+  if (xdgDir) {
+    return join(xdgDir, DIR_PREFIX);
+  }
+  return paths[type];
```

Comment:
minor but don't call `envPaths()` until you need it (meaning, move it after the xdgDir check